### PR TITLE
[openwebnet] Improve shutterRun calibration

### LINKED ISSUE #14215: [openwebnet] Shutter state going from 0 to UNDEF after auto-stop from actuator
## Expected Behavior
If the state is already at 0, the auto-stop should leave it at 0 and not set it back to UNDEF.

## Current Behavior

Test case that **does** lead to the issue:

1. Initial conditions: 
Shutter is up
State is 0
**Shutter Run is 61 seconds**
Auto-stop of actuator is 60 seconds

2. Action: OH is rebooted
Result: state is UNDEF **(OK)**

3. Action: sending an UP command
Result: state changes to 0, but after the auto-stop of the actuator, state goes back to UNDEF. **(NOT OK)**

Here are the log entries of the processing of the auto-stop.

```
2023-01-11 07:15:44.022 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*2*0*51##]
2023-01-11 07:15:44.022 [INFO ] [unication.BUSConnector.message.event] - BUS-MON <<<<<<<< *2*0*51##
2023-01-11 07:15:44.023 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownIdFromMessage(<*2*0*51##>) --> 2.51
2023-01-11 07:15:44.023 [DEBUG] [.handler.OpenWebNetAutomationHandler] - handleMessage(<*2*0*51##>) for thing: openwebnet:bus_automation:22eef285a7:51
2023-01-11 07:15:44.023 [DEBUG] [.handler.OpenWebNetAutomationHandler] - updateAutomationState() - msg=<*2*0*51##> what=STOP
2023-01-11 07:15:44.023 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # current positionEstimation=-1 movedTime=60943
2023-01-11 07:15:44.023 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # movedSteps: 99 UP(-)
2023-01-11 07:15:44.024 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # movingState=0 positionEstimation=-1 - calibrating=-1 shutterRun=61540
2023-01-11 07:15:44.041 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*2*0*51##]
2023-01-11 07:15:44.042 [INFO ] [unication.BUSConnector.message.event] - BUS-MON <<<<<<<< *2*0*51##
2023-01-11 07:15:44.042 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownIdFromMessage(<*2*0*51##>) --> 2.51
2023-01-11 07:15:44.043 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownId=2.51 has NO DEVICE associated, ignoring it
```

Full log file of step 3: [UNDEF_Issue_After_Reboot.txt](https://github.com/openhab/openhab-addons/files/10416507/UNDEF_Issue_After_Reboot.txt)

Test case that **does not** lead to the issue, with down/up initialization:

1. Initial conditions: 
Shutter is up
State is 0
**Shutter Run is 61 seconds**
Auto-stop of actuator is 60 seconds

2. Action: OH is rebooted
Result: state is UNDEF **(OK)**

3. Action: sending a DOWN command, waiting 120 seconds, then sending an UP command
Result: shutter fully closes, then fully opens again, state is 0 **(OK)**

4. Action: sending an UP command
Result: state is 0 and stays that way after auto-stop, the auto-stop is not interfering anymore. **(OK)**

Here are the log entries of the processing of the auto-stop.

```
2023-01-11 15:10:21.287 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*2*0*51##]
2023-01-11 15:10:21.288 [INFO ] [unication.BUSConnector.message.event] - BUS-MON <<<<<<<< *2*0*51##
2023-01-11 15:10:21.288 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownIdFromMessage(<*2*0*51##>) --> 2.51
2023-01-11 15:10:21.289 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownId=2.51 has NO DEVICE associated, ignoring it
2023-01-11 15:10:21.294 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*2*0*51##]
2023-01-11 15:10:21.294 [INFO ] [unication.BUSConnector.message.event] - BUS-MON <<<<<<<< *2*0*51##
2023-01-11 15:10:21.295 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownIdFromMessage(<*2*0*51##>) --> 2.51
2023-01-11 15:10:21.295 [DEBUG] [.handler.OpenWebNetAutomationHandler] - handleMessage(<*2*0*51##>) for thing: openwebnet:bus_automation:22eef285a7:51
2023-01-11 15:10:21.296 [DEBUG] [.handler.OpenWebNetAutomationHandler] - updateAutomationState() - msg=<*2*0*51##> what=STOP
2023-01-11 15:10:21.296 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # current positionEstimation=0 movedTime=60965
2023-01-11 15:10:21.296 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # movedSteps: 99 UP(-)
2023-01-11 15:10:21.296 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # 0 - 99 = -99
2023-01-11 15:10:21.296 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # movingState=0 positionEstimation=0 - calibrating=-1 shutterRun=61540
```

Full log file of step 4: [UNDEF_Issue_After_Initialization.txt](https://github.com/openhab/openhab-addons/files/10416508/UNDEF_Issue_After_Initialization.txt)

Test case that **does not** lead to the issue, with Shutter Run < auto-stop time:

* Initial conditions: 
Shutter is up
State is 0
**Shutter Run is 48 seconds**
Auto-stop of actuator is 60 seconds

* Action: OH is rebooted
Result: state is UNDEF **(OK)**

* Action: sending an UP command
Result: state changes to 0 and stays that way, auto-stop is not interfering **(OK)**

Here are the log entries of the processing of the auto-stop.

```
2023-01-14 04:25:21.528 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*2*0*51##]
2023-01-14 04:25:21.529 [INFO ] [unication.BUSConnector.message.event] - BUS-MON <<<<<<<< *2*0*51##
2023-01-14 04:25:21.530 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownIdFromMessage(<*2*0*51##>) --> 2.51
2023-01-14 04:25:21.530 [DEBUG] [.handler.OpenWebNetAutomationHandler] - handleMessage(<*2*0*51##>) for thing: openwebnet:bus_automation:22eef285a7:51
2023-01-14 04:25:21.530 [DEBUG] [.handler.OpenWebNetAutomationHandler] - updateAutomationState() - msg=<*2*0*51##> what=STOP
2023-01-14 04:25:21.530 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # current positionEstimation=-1 movedTime=60951
2023-01-14 04:25:21.531 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # movedSteps: 127 UP(-)
2023-01-14 04:25:21.531 [DEBUG] [.handler.OpenWebNetAutomationHandler] - # w:51 # movingState=0 positionEstimation=0 - calibrating=-1 shutterRun=48000
2023-01-14 04:25:21.541 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*2*0*51##]
2023-01-14 04:25:21.542 [INFO ] [unication.BUSConnector.message.event] - BUS-MON <<<<<<<< *2*0*51##
2023-01-14 04:25:21.543 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownIdFromMessage(<*2*0*51##>) --> 2.51
2023-01-14 04:25:21.543 [DEBUG] [rnal.handler.OpenWebNetBridgeHandler] - ownId=2.51 has NO DEVICE associated, ignoring it
2023-01-14 04:25:26.571 [INFO ] [nwebnet4j.communication.FrameChannel] - -FC-BUS-MON <------- [*#13**22*05*24*31*000*06*14*01*2023##]
```

Full log file of step 4: [UNDEF_Issue_With_Lower_ShutterRun.txt](https://github.com/openhab/openhab-addons/files/10416525/UNDEF_Issue_With_Lower_ShutterRun.txt)

## Steps to Reproduce (for Bugs)
See above.

## Context
Wrote a script that is sending an UP command to the shutters if winds are getting too strong. Then noticed that after an OH reboot, even with the UP command, the states always went back to UNDEF after 60 seconds, when the auto-stop was sent by the actuator.

## Your Environment
OH 3.4 in a Docker container installed on a Synology NAS.

### PR DESCRIPTION
**Pull request by mvalla**
Impreved shutterRun parameter automatic calibration by subtracting some fiexd (500ms) and relative (2%) timing to the measured shutter run time. Updated README section to explain better shutterRun calibration.
This should fix #14215.

### CODE REVIEW

**Comment by mvalla** [comment]
@lolodomo can you check also this one? thanks!

**Comment by mvalla** [comment]
@lolodobo can you check also this one, left behind? thanks!

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 1]
```diff
@@ -257,31 +257,32 @@ OPEN command to execute: *5*8#134##
...
 
 A powerful feature is to detect scenario activations and CEN/CEN+ buttons press events to trigger rules in openHAB: this way openHAB becomes a very powerful scenario manager activated by BTicino scenario control modules or by CEN/CEN+ scenarios physical buttons.
-See [openwebnet.rules](#openwebnetrules) for examples on how to define rules that trigger on scenarios and on CEN/CEN+ button press events.
+See [openwebnet.rules](#openwebnet-rules) for examples on how to define rules that trigger on scenarios and on CEN/CEN+ button press events.
 
 It's also possible to send _virtual scenario activation_ and _virtual press_ events on the BUS, for example to enable the activation of MH202 or F420 scenarios from openHAB..
-See [openwebnet.sitemap](#openwebnetsitemap) & [openwebnet.rules](#openwebnetrules) sections for examples on how to use the `activateScenario` and `virtualPress` actions connected to a pushbutton on a sitemap.
+See [openwebnet.sitemap](#openwebnet-sitemap) & [openwebnet.rules](#openwebnet-rules) sections for examples on how to use the `activateScenario` and `virtualPress` actions connected to a pushbutton on a sitemap.
```
Comment (lolodomo) [inline bundles/org.openhab.binding.openwebnet/README.md:285]: Links are apparently no more working after these change.

**Review by lolodomo — CHANGES_REQUESTED** [review]
Only few comments from changes in README

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 2]
```diff
@@ -257,31 +257,32 @@ OPEN command to execute: *5*8#134##
...
+- if OH is restarted the binding does not know if a shutter position has changed in the meantime, so its position will be `UNDEF`. Move the shutter all `UP`/`DOWN` to synchronise again its position with the binding
+- the shutter position is estimated based on UP/DOWN timings: an error of ±2% is normal
 
 #### Scenario channels
 
 Basic Scenarios and CEN/CEN+ Scenarios channels are [TRIGGER channels](https://www.openhab.org/docs/configuration/rules-dsl.html#channel-based-triggers]): they handle events and do not have a state.
 
 A powerful feature is to detect scenario activations and CEN/CEN+ buttons press events to trigger rules in openHAB: this way openHAB becomes a very powerful scenario manager activated by BTicino scenario control modules or by CEN/CEN+ scenarios physical buttons.
-See [openwebnet.rules](#openwebnetrules) for examples on how to define rules that trigger on scenarios and on CEN/CEN+ button press events.
+See [openwebnet.rules](#openwebnet-rules) for examples on how to define rules that trigger on scenarios and on CEN/CEN+ button press events.
```
[the remark this answers is not in the document]
Comment (mvalla) [reply bundles/org.openhab.binding.openwebnet/README.md:282]: @lolodomo there are several MarkDown dielects on how to write links properly when there are special chars in the link.
What is the right syntax that is used in the MD render used for rendering READMEs ?

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.openwebnet/README.md:282]: I don't know, I just tested your change and it is not working.
Just go in "Files changed" and do "View file" on README.md. Try your changed links, there are not working. Try an unchanged link and it works.
Now go in Git on README.md from the main branch and try the original links (before you changed them) and they works.
BUT if I display the documentation from openhab.org and try the links (those you want to change), they are not working !

@stefan-hoehn : any idea about that ?

File: bundles/org.openhab.binding.openwebnet/README.md [hunk 3]
```diff
@@ -257,31 +257,32 @@ OPEN command to execute: *5*8#134##
...
-- before adding/configuring roller shutter Things it is suggested to have all roller shutters `UP`, otherwise the Percent command won’t work until the roller shutter is fully rolled up
-- if OH is restarted the binding does not know if a shutter position has changed in the meantime, so its position will be `UNDEF`. Move the shutter all `UP`/`DOWN` to synchronize again its position with the binding
-- the shutter position is estimated based on UP/DOWN timing: an error of ±2% is normal
+- if `shutterRun` is wrongly set *higher* than the actual runtime or the "stop time" of the actuator (see NOTE above), then position estimation will remain `UNDEF`: try to reduce `shutterRun` until you find the right value
+- before adding/configuring roller shutter Things it is suggested to have all roller shutters `UP`, otherwise the Percent command will not work until the roller shutter is fully rolled up
+- if OH is restarted the binding does not know if a shutter position has changed in the meantime, so its position will be `UNDEF`. Move the shutter all `UP`/`DOWN` to synchronise again its position with the binding
```
[the remark this answers is not in the document]
Comment (mvalla) [reply bundles/org.openhab.binding.openwebnet/README.md]: done

File: bundles/org.openhab.binding.openwebnet/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvalla) [reply bundles/org.openhab.binding.openwebnet/README.md]: I checked on VSC .md preview tab and in fect the previous version works, so I reveretd to it

File: bundles/org.openhab.binding.openwebnet/README.md (hunk 1 above)
Comment (mvalla) [reply bundles/org.openhab.binding.openwebnet/README.md]: reverted to previous version

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**